libs/brig-types/src/Brig/Types/TURN.hs

```diff
@@ -258,3 +264,65 @@ instance ToJSON Transport where
 -- Convenience
 optional :: (Alternative f, Functor f) => f a -> f (Maybe a)
 optional x = option Nothing (Just <$> x)
+
+-- | given a list of URIs and a size, limit URIs
+-- with order priority from highest to lowest: UDP -> TLS -> TCP
+-- i.e. (if enough servers of each type are available)
+--   1 -> 1x UDP
+--   2 -> 1x UDP, 1x TLS
+--   3 -> 1x UDP, 1x TLS, 1x TCP
+--   4 -> 2x UDP, 1x TLS, 1x TCP
+--   5 -> 2x UDP, 2x TLS, 1x TCP
+--    ... etc
+-- if not enough servers are available, prefer udp, then tls
+limitServers :: [TurnURI] -> Int -> [TurnURI]
```

Comment:
Not sure you prefer this "style" but since you asked :)
```
limitServers :: [TurnURI] -> Int -> [TurnURI]
limitServers uris = limitHeuristic [] (splitByType ([],[],[],[]) uris)
  where
    splitByType acc       [] = acc
    splitByType (a,b,c,d) (x:xs)
        | isUdp x   = splitByType (x:a, b  , c  ,  d) xs
        | isTls x   = splitByType (a  , x:b, c  ,  d) xs
        | isTcp x   = splitByType (a  , b  , x:c,  d) xs
        | otherwise = splitByType (a  , b  , c  ,x:d) xs
```

Reply:
Ah, thanks! Yes, I prefer that style.